src/nav/navReducers.js

```diff
@@ -21,11 +21,22 @@ import {
   LOGOUT,
   SWITCH_NARROW,
 } from '../actionConstants';
+import { getAuth } from '../account/accountSelectors';
+import { getAccounts } from '../directSelectors';
 
 const initialState = getStateForRoute('loading') || NULL_NAV_STATE;
 
-const rehydrate = (state: NavigationState, action: RehydrateAction): NavigationState =>
-  getInitialNavState(action.payload) || state;
+const rehydrate = (state: NavigationState, action: RehydrateAction): NavigationState => {
+  if (!getAuth(action.payload).apiKey) {
+    const accounts = getAccounts(action.payload);
+    // getStateForRoute can return null, but it is unclear under what
+    // conditions. Empirically, it doesn't return null on the initial start of
+    // the app, but this should be verified.
+    // $FlowFixMe: getStateForRoute may return null but it shouldn't.
+    return getStateForRoute(accounts && accounts.length > 1 ? 'account' : 'welcome');
```

Comment:
@gnprice I saw that in 9266bff170ee9cad0622c87885787b0717172024 you changed the behavior of getStateForRoute to return `null` when getActionForPathAndParams returns null. Do we know under what circumstances getActionForPathAndParams returns null?